google/logging_utils.go

```diff
@@ -0,0 +1,59 @@
+package google
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// loggingSinkTypes contains all the possible Stackdriver Logging resource types. Used to parse ids safely.
+var loggingSinkTypes = []string{
+	"billingAccount",
+	"folders",
+	"organizations",
+	"projects",
+}
+
+// LoggingSinkId represents the parts that make up the canonical id used within terraform for a logging resource.
+type LoggingSinkId struct {
+	typ     string
+	typName string
```

Comment:
In the docs, the `parent` parameter documentation says: 
```md
The resource in which to create the sink:
"projects/[PROJECT_ID]"
"organizations/[ORGANIZATION_ID]"
"billingAccounts/[BILLING_ACCOUNT_ID]"
"folders/[FOLDER_ID]"
```

Instead of type, we could use resource. 

resourceType instead of typ
resourceId instead of typName

Reply:
Your names are much better! Fixed.